Update readmes for Image Prioritizer, Embed Optimizer, and Optimization Detective (#1636)
WordPress/performance: plugins/image-prioritizer/readme.txt

== Description ==

> This plugin optimizes the loading of images (and videos) with prioritization, lazy loading, and even some size reduction:

Reviewer: Two things regarding this:
* "even some size reduction" can be misunderstood that the plugin may reduce file size of images, which of course it does not. I would rephrase this to be about "size selection" rather than "size reduction".
* I think this sentence should end in a period, as sort of a short description, and then you can use "The current optimizations include:", just how it is for Embed Optimizer.
```suggestion
This plugin optimizes the loading of images (and videos) with prioritization, lazy loading, and more accurate image size selection.

The current optimizations include:
```

2. Adding breakpoint-specific `fetchpriority=high` preload links for the LCP elements which are `IMG` elements or elements with a CSS `background-image` inline style.
3. Applying lazy-loading to `IMG` tags based on whether they appear in any breakpoint’s initial viewport. (Additionally, [`sizes=auto`](https://make.wordpress.org/core/2024/10/18/auto-sizes-for-lazy-loaded-images-in-wordpress-6-7/) is then also correctly applied.)
4. Adding `fetchpriority=low` to `IMG` tags which appear in the initial viewport but are not visible, such as when they are subsequent carousel slides.
5. Reducing the size of the `poster` image of a `VIDEO` from full size to the size appropriate for the maximum width of the video (on desktop).
6. Lazy-loading `VIDEO` tags by setting the appropriate attributes based on whether they appear in the initial viewport. If a `VIDEO` is the LCP element, it gets `preload=auto`; if it is in an initial viewport, the `preload=metadata` default is left; if it is not in an initial viewport, it gets `preload=none`. Lazy-loaded videos also get initial `preload`, `autoplay`, and `poster` attributes restored when the `VIDEO` is going to enter the viewport.

> This plugin requires the [Optimization Detective](https://wordpress.org/plugins/optimization-detective/) as a dependency. Please refer to that plugin for additional background on how this plugin works as well as additional developer options. Information about images and videos displayed on a given visit are stored in URL Metrics grouped by the following viewport widths according to WordPress’s common responsive breakpoints:

Reviewer: Missing "plugin" word, also I'd suggest to make this bold like you did for Embed Optimizer.

For content, I think this has too many details about Optimization Detective, e.g. the breakpoints are defined by Optimization Detective, so such information should be in the readme of Optimization Detective, which is already referred to here for more information. Otherwise, such information would be just as appropriate to have in the Embed Optimizer readme, at which point we are duplicating information.
```suggestion
**This plugin requires the [Optimization Detective](https://wordpress.org/plugins/optimization-detective/) plugin as a dependency.** Please refer to that plugin for additional background on how this plugin works as well as additional developer options.
```

Mobile: 0-480px  
Phablet: 481-600px  
Tablet: 601-782px  
Desktop: \>782px

> 

Reviewer: See above:
```suggestion
```

This grouping is essential for proper optimization since different breakpoints in a theme's responsive design may result in differing elements being the LCP element or whether an element is visible at all.

> 

Reviewer: See above:
```suggestion
```

> 👉 **Note:** This plugin optimizes pages for actual visitors, and it depends on visitors to optimize pages. As such, you won't see optimizations applied immediately after activating the plugin. Please wait for URL Metrics to be gathered for both mobile and desktop visits. And since administrator users are not normal visitors typically, optimizations are not applied for admins by default.

Reviewer: Can you add a similar note to the Optimization Detective specific section in the Embed Optimizer readme?

Author: I think there is already such a note? See:

https://github.com/WordPress/performance/blob/9d9136645cd94734e296dadf336473333f7a4f3f/plugins/embed-optimizer/readme.txt#L28

Reviewer: Ah missed that, thanks!
